server.go

```diff
@@ -115,6 +121,71 @@ func ReadOnly() ServerOption {
 	}
 }
 
+// RequestType is the type of Request a client made.
+type RequestType uint
+
+const (
+	Open RequestType = iota
+	Close
+	Read
+	Write
+	Lstat
+	Fstat
+	Setstat
+	Fsetstat
+	Opendir
+	Readdir
+	Remove
+	Mkdir
+	Rmdir
+	Realpath
+	Stat
+	Rename
+	Readlink
+	Symlink
+)
+
+// RequestPacket is information about a client request.
+type RequestPacket struct {
+	Type RequestType
+	// Path is the path the request specified, or if the request specified
+	// a handle instead, Path is the path corresponding to that handle.
+	Path string
+	// TargetPath is the new path in a rename request, or the new path in
+	// a symlink request.
+	TargetPath string
+	// Flags is any flags the request passed.
+	Flags      uint32
+	Attributes *Attributes
+	// Err is the error that occured from handling the request.
+	Err error
+}
```

Comment:
This appears to be recreating the already existing [sftp.Request](https://pkg.go.dev/github.com/pkg/sftp#Request) struct.

What does this purposed API provide that cannot already be served via the `RequestServer` abstraction? I’m quite loathe to build in and export Yet Another Abstraction that fits no better the SSH_FILEXFER standards it is supposed to be representing.